scripts/van_allen_tools/migrate_van_allen_workspaces.py

```diff
@@ -202,49 +199,231 @@ def setup_single_workspace(workspace):
+    destination_workspace_name = migration_data["destination_workspace_name"]
+    destination_workspace_namespace = migration_data["destination_workspace_namespace"]
+
+    # Set up the migration_data dict
+    migration_data_full = migration_data
+
+    # Default copy_successful
+    copy_successful = True
+
+    # Default Error
+    error = "NA"
+
+    # Get the list of all the workflows
+    try:
+        workflow_list = fapi.list_workspace_configs(source_workspace_namespace, source_workspace_name)
+
+        for workflow in workflow_list.json():
+            # Get workflow config (overview config)
+            workflow_config = workflow['methodRepoMethod']
+
+            # Get workspace config (Detailed config with inputs, oututs, etc)
+            workspace_config = fapi.get_workspace_config(source_workspace_namespace, source_workspace_name, workflow_config['methodNamespace'], workflow_config['methodName'])
+
+            # Create a workflow based on Detailed config
+            fapi.create_workspace_config(destination_workspace_namespace, destination_workspace_name, workspace_config.json())
+            print(f"Copied {workflow_config['methodName']} workflow : over to {destination_workspace_namespace}/{destination_workspace_name}")
+    except Exception as e:
+        copy_successful = False
+        print(f"Error: {e}")
+        error = f"Error: {e}"
+
+    # Adding column to tsv
+    migration_data_full["copy_workflow"] = copy_successful
+    migration_data_full["workflow_error"] = error
+
+    return migration_data_full
+
+
+def migrate_workspaces(tsv):
+    """Create and set up migrated workspaces."""
     # read full tsv into dataframe
     setup_info_df = pd.read_csv(tsv, sep="\t")
 
     # create df for output tsv file
-    col_names = ["original_workspace_name", "original_workspace_namespace",
-                 "new_workspace_name", "new_workspace_namespace",
-                 "workspace_link", "workspace_bucket",
+    col_names = ["source_workspace_name", "source_workspace_namespace", 
+                 "source_workspace_bucket", "destination_workspace_name", "destination_workspace_namespace",
+                 "destination_workspace_bucket", "workspace_link",
                  "workspace_creation_error",
                  "workspace_ACLs", "workspace_ACLs_error",
                  "workspace_tags", "workspace_tags_error",
-                 "final_workspace_status"]
+                 "final_workspace_status", "copy_data_table", 
+                 "data_table_error", "copy_workflow", "workflow_error"]
 
     all_row_df = pd.DataFrame(columns=col_names)
 
     # per row in tsv/df
     for index, row in setup_info_df.iterrows():
 
-        row_dict = setup_single_workspace(row)
-        all_row_df = all_row_df.append(row_dict, ignore_index=True)
+        # Create Workspace
+        migration_data = setup_single_workspace(row)
+
+        # Copy over data table
+        migration_data_with_dt = copy_workspace_entities(migration_data)
```

Comment:
moving up the data table migration/copying into `setup_single_workspace` the way we did `copy_workspaces` together.